libs/remix-core-plugin/src/lib/compiler-artefacts.ts

```diff
@@ -72,51 +72,61 @@ export class CompilerArtefacts extends Plugin {
-      if(dirList.includes(path + '/artifacts')) {
-        const fileList = await this.call('fileManager', 'fileList', path + '/artifacts')
-        const artefactsFilePaths = fileList.filter(filePath => {
-          const filenameArr = filePath.split('/')
-          const filename = filenameArr[filenameArr.length - 1]
-          if (filename === `${contractName}.json` || filename === `${contractName}_metadata.json`) return true
-        })
-        if (artefactsFilePaths && artefactsFilePaths.length) {
-          const content = await this.call('fileManager', 'readFile', artefactsFilePaths[1])
-          const artifacts = JSON.parse(content)
-          return { abi: artifacts.abi, bytecode: artifacts.data.bytecode.object }
-        } else {
-          for (const dirPath of dirList) {
-            const result = await this.getArtefactsFromFE (dirPath, contractName)
-            if (result) return result
+      for (const dirPath of dirList) {
+        if(dirPath === path + '/artifacts' && await this.call('fileManager', 'exists', dirPath + '/build-info')) {
+          const buildFileList = await this.call('fileManager', 'fileList', dirPath + '/build-info')
+          for (const buildFile of buildFileList) {
+            let content = await this.call('fileManager', 'readFile', buildFile)
+            if (content) content = JSON.parse(content)
+            const contracts = content.output.contracts
+            const artefacts = this._getAllContractArtefactsfromOutput(contracts, contractName)
+            Object.assign(contractArtefacts, artefacts)
           }
-        }
-      } else {
-        for (const dirPath of dirList) {
-          const result = await this.getArtefactsFromFE (dirPath, contractName)
-          if (result) return result
-        }
-      }
+        } else await this._getArtefactsFromFE (dirPath, contractName, contractArtefacts)
+      } 
     } else return
   }
 
   async getArtefactsByContractName (contractName) {
     const contractsDataByFilename = this.getAllContractDatas()
-    const contractsData = Object.values(contractsDataByFilename)
-    if (contractsData && contractsData.length) {
-      const index = contractsData.findIndex((contractsObj) => Object.keys(contractsObj).includes(contractName))
-      if (index !== -1) return { abi: contractsData[index][contractName].abi, bytecode: contractsData[index][contractName].evm.bytecode.object }
+    if (contractName.includes(':')) {
+      const nameArr = contractName.split(':')
+      const filename = nameArr[0]
+      const contract = nameArr[1]
+      if(Object.keys(contractsDataByFilename).includes(filename) && contractsDataByFilename[filename][contract]) 
+        return contractsDataByFilename[filename][contract]
       else {
-        const result = await this.getArtefactsFromFE ('contracts', contractName)
-        if (result) return result
+        const allContractsData = {}
+        await this._getArtefactsFromFE ('contracts', contract, allContractsData)
```

Comment:
having a return value isn't possible in that case? using the spread feature

Reply:
Complete data is populated in `allContractsData`

Reply:
yes I see, I am just wondering why you do like this instead of having `_getArtefactsFromFE` to return something.
but ok I guess that is fine like that, could you just rename the function `_populateContractsArtefactsFromFE`

Reply:
done